TestFoundation/TestNSData.swift

```diff
@@ -908,6 +910,45 @@ extension TestNSData {
         XCTAssertTrue(didCatchError)
     }
 
+    func test_contentsOfFile() {
+        let testDir = testBundle().resourcePath
+        let filename = testDir!.appending("/NSStringTestData.txt")
+
+        let contents = NSData(contentsOfFile: filename)
+        XCTAssertNotNil(contents)
+        if let contents = contents {
+            let ptr =  UnsafeMutableRawPointer(mutating: contents.bytes)
+            let str = String(bytesNoCopy: ptr, length: contents.length,
+                         encoding: .ascii, freeWhenDone: false)
+            XCTAssertEqual(str, "swift-corelibs-foundation")
+        }
+    }
+
+    func test_contentsOfZeroFile() {
+#if os(Linux)
+        let contents = NSData(contentsOfFile: "/proc/self/cmdline")
```

Comment:
It's possible that the `/proc/` filesystem isn't mounted in some situations, so I would guard this with a check to see if `/proc/self/cmdline` exists and silently abort if it doesn't - because it's possible someone will run this with a minimal container where procfs isn't present.